R/odbc-connection.R

```diff
@@ -460,3 +460,56 @@ setMethod("odbcConnectionTableTypes", "OdbcConnection",
     connection_sql_table_types(conn@ptr)
   }
 )
+
+
+# Internal method, used to check for target existence
+# in dbWriteTable.  Historically we used DBI::dbExistsTable
+# however its mission may be interpreted more broadly
+# ( does this table exists period ) versus the question
+# we are trying to answer in dbWriteTable ( does this
+# table identifier I may be writing to exist? )  In
+# particular, the former may mean, across catalogs or schema;
+# the latter we understand to mean:
+# 1. At the specified catalog/schema; if not specified then
+# 2A. In the current catalog/schema; OR
+# 2B. In a temporary table.
+#
+# While by default we continue to route requests to dbExistsTable,
+# this narrower interpretation may allow some back-ends
+# to optimize this search.
+setGeneric(
+  "dbExistsTableForWrite",
+  valueClass = "logical",
+  function(conn, name, ...) {
+    standardGeneric("dbExistsTableForWrite")
+  }
+)
+
+#' @rdname OdbcConnection
```

Comment:
Since this is our own generic and this help-file is described as

> Implementations of pure virtual functions defined in the DBI package for OdbcConnection objects.

I might just `@noRd` these methods.